src/lib/checkpoint/cp_mgr.cpp

```diff
@@ -35,6 +36,9 @@ CPManager::CPManager() :
         [this](meta_blk* mblk, sisl::byte_view buf, size_t size) { on_meta_blk_found(std::move(buf), (void*)mblk); },
         nullptr);
 
+    resource_mgr().register_dirty_buf_exceed_cb(
```

Comment:
Should we wait for cp flush to complete in this exceeded size case ?

Reply:
cp manager controls this, right? If a cp is running, a back-2-back cp will be created.

Reply:
Q: will `dirty_buf_exceed_cb` every time a new dirty_buf added after exceeding the limit?
Trying to avoid the case that we exceeded the dirty limit for a moment and a lot of unnecessary CP get triggered.

Reply:
CP manager will garunteen only one cp can be running at anytime. e.g. if a cp is already running, it returns immediately. And if consumer asks for a forced trigger_cp, a back-2-back cp will be created.  
If cp is slow or stuck (due to bugs or disk issues), it push the back pressure to consumers, e.g. no more dirty buffer can be allocated and push back I/O failures. In HomeStore 4.x, I think what we